opensfm/pairs_selection.py

```diff
@@ -558,6 +567,8 @@ def match_candidates_by_order(
 
     pairs = set()
     for i, image_ref in enumerate(images_ref):
+        if data.matches_exists(image_ref):
```

Comment:
Have you tested this in a data set of 2 images.

    Run match
    Add 3rd image
    Run match again
    Observe changes to the matches files?